src/scaffold/store.ts

```diff
@@ -45,17 +45,20 @@ export const useStore = create<Store>((set) => {
 
       case 'sendTemplatesOptions':
         const templatesOptions = message.templatesOptions;
-        if (!templatesOptions) {
-          dispatch(set, (state) => ({
-            ...state,
-            templateOptions: [],
-          }));
-          return;
+        const values: Record<string, string> = {};
+        if (templatesOptions) {
+          Object.keys(templatesOptions).forEach((option) => {
+            values[option] = templatesOptions[option].default;
+          });
         }
-        const options = Object.keys(templatesOptions).map((key) => [key, templatesOptions[key]]);
         dispatch(set, (state) => ({
           ...state,
-          templateOptions: options,
+          isLoading: false,
+          templateOptions: templatesOptions ?? {},
```

Comment:
Just gather the `templateOptions` like

```js
const { templateOptions = {} } = message;
```

then you'll know that `templateOptions` always exist. The `if` and this code is then unnecessary.

Reply:
Why do you mark it as `resolved`. Apparently its not resolved.